types/index.d.ts

```diff
@@ -0,0 +1,12 @@
+import {ChartType, Plugin} from 'chart.js';
+import {Options} from './options'
+
+declare module 'chart.js' {
+  interface PluginOptionsByType<TType extends ChartType> {
+    deferred?: Options
+  }
+}
+
+declare const deferred: Plugin;
```

Comment:
I would rename `deferred` to `plugin` to be consistent with [this declaration](https://github.com/chartjs/chartjs-plugin-datalabels/blob/master/types/index.d.ts#L22).

Reply:
I took inspiration from the [annotation plugin](https://github.com/chartjs/chartjs-plugin-annotation/blob/ccc89987a42c63f0c84d012e2fbdcc6e119ed06c/types/index.d.ts#L20) where they do it like that.

So it will always be inconsistent with one of the other unless either the datalabels or annotation makes it also differently

Reply:
True and it doesn't have an impact on the actual exported name because it's `default`. However, since I'm maintaining the `datalabels` and `deferred` plugins, I would prefer both to be consistent.